### ISSUE TITLE
linter: disable directives fail when trailing explanations are applied

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#22152]
### Summary

Disable directives can stop suppressing diagnostics when an explanation is added after the rule name. In the latest commit, several comments had to be changed from:

```ts
// oxlint-disable-next-line typescript/unbound-method - bound in `runHook`
```

to:

```ts
// oxlint-disable-next-line typescript/unbound-method
```

because the trailing explanation caused the directive to no longer match `typescript/unbound-method`.

### Reproduction

```ts
class C {
  method() {}
}

const c = new C();

// oxlint-disable-next-line typescript/unbound-method - bound later
const method = c.method;
```

Run:

```sh
oxlint --type-aware --type-check -D typescript/unbound-method repro.ts
```

### Expected Behavior

The directive should suppress `typescript/unbound-method`, or oxlint should clearly document and/or diagnose unsupported trailing explanation syntax.

ESLint-style descriptions commonly use `--`, for example:

```ts
// oxlint-disable-next-line typescript/unbound-method -- bound later
const method = c.method;
```

That form should continue to work consistently across Rust linting, JS plugin APIs, unused-disable reporting, and autofixes.

### Actual Behavior

With a trailing single-hyphen explanation, oxlint treats the explanation as part of the rule name, so the disabled rule becomes something like:

```text
typescript/unbound-method - bound later
```

That does not match the emitted `typescript/unbound-method` diagnostic, so the warning/error is still reported.

**Comment by kirillgroshkov** [comment]
Adding a related case under the same root cause:

Multiple space-separated rule names in a disable directive also stop working in 1.63.0:

```js
// oxlint-disable-next-line no-bitwise no-implicit-coercion
const x = (1 >> 0) === +'1'
```

- 1.62.0 → 0 errors (both rules suppressed) ✅
- 1.63.0 → 2 errors (directive entirely ignored) ❌

Comma-separated form (`no-bitwise, no-implicit-coercion`) still works.

Same root cause as this issue: PR #21906 swapped `name.contains(rule_name)` for an exact-equality check, but the directive parser still stores the whole rule-list tail as a single name string. Any tail that isn't a single rule name — multi-rule lists with whitespace separators, or trailing `- description` explanations — no longer matches.

Both ESLint and prior oxlint versions supported the space-separated form, so this is a backwards-compat break worth flagging alongside the trailing-explanation case.

**Comment by camc314** [comment]
@kirillgroshkov apologies for the break, should be fixed in the next release.
